File: 무개과제/알고리즘/001. 추억은 BubbleBubble/추억은 BubbleBubble.kt
// 오름차순 정렬
fun bubbleSort(arr: IntArray): IntArray {
    for (r in arr.size - 1 downTo 1) {
        var count = 0

        for (i in 0 ..< r) {
            if (arr[i] > arr[i + 1]) {
                arr[i] = arr[i + 1].also { arr[i + 1] = arr[i] }
                count++
            }
            println(arr.toList())
        }
        if(count == 0) {
            break
        }
    }
    return arr
}

File: 알고리즘/무개과제/001. 추억은 BubbleBubble/추억은 BubbleBubble.py
def bubble_sort(arr):
    for r in range(len(arr)):
        changed = False
        for i in range(0, len(arr) - r - 1):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                changed = True
        if not changed:
            break
    return arr
